src/realm/CMakeLists.txt

```diff
@@ -213,10 +213,24 @@ elseif(ANDROID)
     list(APPEND REALM_EXTRA_LIBS atomic)
 endif()
 
-# The placeholder file is needed in order to make builds work on Xcode
-# The issue is described in the last sentence here:
-# https://cmake.org/cmake/help/v3.7/command/add_library.html#object-libraries
-add_library(realm STATIC $<TARGET_OBJECTS:realm-objects> placeholder.cpp)
+set(REALM_XCODE_DEPENDENCY "")
+if(CMAKE_GENERATOR STREQUAL Xcode)
+    # The placeholder file is needed in order to make builds work on Xcode
+    # The issue is described in the last sentence here:
+    # https://cmake.org/cmake/help/v3.7/command/add_library.html#object-libraries
+    # Another problem is that Xcode does not see changes in the realm-objects
+    # target, to solve this we trigger rebuiliding the library by making
+    # placeholder.cpp depend on all the sources rather than realm-objects.
+    set(REALM_XCODE_DEPENDENCY "placeholder.cpp")
+    add_custom_command(
+        OUTPUT ${REALM_XCODE_DEPENDENCY}
+        DEPENDS ${REALM_SOURCES} ${REALM_INSTALL_ALL_HEADERS}
+        COMMAND echo \"//This file is generated by cmake for Xcode builds\" > ${REALM_XCODE_DEPENDENCY}
+        COMMAND echo \"extern int xcode_unused\;\" >> ${REALM_XCODE_DEPENDENCY}
```

Comment:
One thing I noticed while playing around with this branch was that if you build the default `ALL_BUILD` target in Xcode, placeholder.cpp typically ends up containing the following:
```
//This file is generated by cmake for Xcode builds
extern int xcode_unused;
extern int xcode_unused;
```

I suspect this is because the `realm` and `realm-shared` targets both independently run the command to generate placeholder.cpp at the same time, resulting in both contributing to the final file.

This doesn't appear to create any obvious problems, but I thought I'd make a note of it here in case we see anything surprising that could perhaps be explained by this.